src/test/scala/com/fulcrumgenomics/bam/FindSwitchbackReadsTest.scala

```diff
@@ -0,0 +1,350 @@
+    rec.bases(10) = 'a'
+    FindSwitchbackReads.findReadBasedSwitchback(refMap, template, minLength=5, maxOffset=20, maxErrorRate=0.1).isDefined shouldBe true
+    rec.bases(15) = 'c'
+    FindSwitchbackReads.findReadBasedSwitchback(refMap, template, minLength=5, maxOffset=20, maxErrorRate=0.1).isDefined shouldBe true
+    rec.bases(20) = 't'
+    FindSwitchbackReads.findReadBasedSwitchback(refMap, template, minLength=5, maxOffset=20, maxErrorRate=0.1).isDefined shouldBe true
+    rec.bases(25) = 'g'
+    FindSwitchbackReads.findReadBasedSwitchback(refMap, template, minLength=5, maxOffset=20, maxErrorRate=0.1).isDefined shouldBe false
+  }
+
+  it should "find a switchback in a read when given a template with paired reads" in {
+    val builder  = new SamBuilder(readLength=50, sd=Some(ref.getSequenceDictionary))
+    val recs     = builder.addPair(contig=0, start1=1,  cigar1="50M",    strand1=Plus,  bases1="TGGCACCGAGCGAGCCGCGATGACAATGGCTGCATTGTGCTTCATGTCCC",
+                                             start2=71, cigar2="40M10S", strand2=Minus, bases2="CTGGACTCCTTCCACTCGCGGGTCGTCTCCAGAGCTCAGAtctgagctct")
+    val template = Template(recs.iterator)
+    val hit      = FindSwitchbackReads.findReadBasedSwitchback(refMap, template, minLength=5, maxOffset=20, maxErrorRate=0.1)
+    hit.value.length shouldBe 10
+    hit.value.offset shouldBe 0
+    hit.value.read shouldBe recs.find(_.secondOfPair)
+  }
+
+  it should "not find switchbacks in secondary or supplementary records" in {
+    val builder  = new SamBuilder(readLength=50, sd=Some(ref.getSequenceDictionary))
+    builder.addFrag(name="q1", contig=0, start=71, cigar="50M", strand=Plus, bases="CTGGACTCCTTCCACTCGCGGGTCGTCTCCAGAGCTCAGAAAATGAGGTG")
+    builder.addFrag(name="q1", contig=0, start=71, cigar="10S40M", strand=Plus, bases="aggagtccagCTGGACTCCTTCCACTCGCGGGTCGTCTCCAGAGCTCAGA").foreach(_.secondary = true)
+    builder.addFrag(name="q1", contig=0, start=71, cigar="12S38M", strand=Plus, bases="gaaggagtccagCTGGACTCCTTCCACTCGCGGGTCGTCTCCAGAGCTCA").foreach(_.supplementary = true)
+    val template = Template(builder.iterator)
+    val hit      = FindSwitchbackReads.findReadBasedSwitchback(refMap, template, minLength=5, maxOffset=20, maxErrorRate=0.1)
+    hit shouldBe None
+  }
+
+  "FindSwitchbackReads.findTandemSwitchback" should "call FF and RR templates with small enough gaps as switchbacks" in {
+    val builder = new SamBuilder(readLength=50, sd=Some(ref.getSequenceDictionary))
+    val ff = Template(builder.addPair(contig=0, start1=100, start2=160, strand1=Plus, strand2=Plus).iterator)
+    val rr = Template(builder.addPair(contig=0, start1=100, start2=160, strand1=Minus, strand2=Minus).iterator)
+
+    FindSwitchbackReads.findTandemSwitchback(ff, maxGap=200).value.gap shouldBe 10
+    FindSwitchbackReads.findTandemSwitchback(rr, maxGap=200).value.gap shouldBe 10
+  }
+
+  it should "not call tandem pairs switchbacks if the gap is too big" in {
```

Comment:
```suggestion
  it should "not call pairs tandem switchbacks if the gap is too big" in {
```

Reply:
I prefer the original wording.  They are 100% "tandem pairs" and the outcome I want is that they are identified as switchbacks via the tandem-finding method.